### ISSUE TITLE
Using with Github Pages?

### ISSUE THREAD

**Issue by zackhall** [sverrirs/jekyll-paginate-v2#9]
Hi –

First of all, thank you for taking the time to build this. I found it very useful. I'm sorry to open this as an issue, since it's more of a question. 

I used this locally and it worked perfectly. When I deployed it to gh-pages though on my site, zthall.com, the zthall.com/blog page is now blank. Any ideas why this wouldn't work with gh-pages?

**Comment by zackhall** [comment]
Here's the page that I use it on – it's not on the homepage, but on a subpage at /blog. 

https://github.com/zackhall/pixyll/blob/gh-pages/blog.html

**Comment by sverrirs** [comment]
Hi Zack, 
Thanks for the kind words, glad you found the gem useful. 

Unfortunately the gem is not currently supported by github-pages. I've had brief discussion with @parkr about the possibility of taking over the old plugin but we did not discuss anything regarding github. But if the github people will consider it good enough to include on their system I would  be honoured. 

So the current work around is to generate the site locally and upload the _site folder to github. sorry.

**Comment by zackhall** [comment]
> GitHub Pages officially supports the Jekyll plugins found in the GitHub Pages gem. For the exact versions of the Jekyll plugins that GitHub Pages supports, see this list of GitHub Pages dependencies. Other plugins are not supported, so the only way to incorporate them in your site is to generate your site locally and then push your site's static files to your GitHub Pages site.

:-(

**Comment by zackhall** [comment]
Woah, nice timing! 👍 Thanks for your help, much appreciated. And hopefully someday this will supplant the old plugin.

**Comment by justinechacko** [comment]
Hi, I am a newbie. I dont understand the conversation. Can I use it in gitlab or github?

**Comment by sverrirs** [comment]
Please note that this plugin is currently NOT supported by GitHub pages. Here is a [list of all plugins supported](https://pages.github.com/versions/). There is work underway to try to get it added it but until then please follow [this GitHub guide](https://help.github.com/articles/adding-jekyll-plugins-to-a-github-pages-site/) to enable it or use [Travis CI](https://ayastreb.me/deploy-jekyll-to-github-pages-with-travis-ci/).

GitLab supposedly supports [any plugin](https://about.gitlab.com/comparison/gitlab-pages-vs-github-pages.html)

**Comment by UnderpantsGnome** [comment]
If you want to use this pagination and the GitHub pages gem, though I'm guessing it won't build on GitHub pages, you can use this [fork](https://github.com/UnderpantsGnome/pages-gem) of pages-gem as long as you are using a Gemfile.

The Gemfile needs to look like this (plus whatever other gems you are using)

    source 'https://rubygems.org'

    git_source(:github) do |repo_name|
      repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
      "https://github.com/#{repo_name}.git"
    end

    gem "github-pages",
      github: 'UnderpantsGnome/pages-gem',
      branch: "paginate-v2",
      group: :jekyll_plugins

This is also relevant for #82 or possibly more so than this issue.

The commit that enables this https://github.com/UnderpantsGnome/pages-gem/commit/127c31b74e4bd088fab81677a66bb9769162f631

**Comment by jzeneto** [comment]
Hi, is there any advancements in making gh-pages support this plugin instead of the old (and *discontinued*) plugin?

**Comment by AlexSikilinda** [comment]
@parkr, @sverrirs it would be gread to have this plugin supported by github pages. Is there any reason it is not added?

**Comment by sverrirs** [comment]
Hi all,

From my side there is nothing blocking. Perhaps @parkr can shed some light on the github side and if there is some blocking issues on their end I can prioritize fixing them as soon as possible. :)

**Comment by parkr** [comment]
@sverrirs My worry is causing user confusion. Having a separate gem which may conflict with the Jekyll-paginate gem is not ideal to me. How would you feel merging your code with `jekyll/jekyll-paginate` and ensuring it is backwards-compatible? Ideally we'd release a v2.0 of jekyll-paginate which is still backwards-compatible and GitHub Pages could use that merged version.

**Comment by parkr** [comment]
@sverrirs My worry is causing user confusion. Having a separate gem which may conflict with the Jekyll-paginate gem is not ideal to me. How would you feel merging your code with `jekyll/jekyll-paginate` and ensuring it is backwards-compatible? Ideally we'd release a v2.0 of jekyll-paginate which is still backwards-compatible and GitHub Pages could use that merged version.

**Comment by sverrirs** [comment]
Hi @parkr :) 

I have no objections to merging this implementation with the old one. I agree that having the two versions is likely to confuse. 

I'm wondering if you would give me access to admin the old project in the samewo after the move? 

Also, could I expect the project to be somehow supported by the jekyll core team? The reason I ask is because I'm currently looking for someone to assist me with the maintenance and co-administration of this repo and would hate to see it go abandoned if we migrate.

**Comment by ashmaroli** [comment]
@sverrirs Hi! :wave:
I'm part of the team that maintains plugins at the Jekyll Org.

* All plugins at the Jekyll Org are regularly updated as necessary.
* All plugins follow more or less the same "code style" used in Jekyll gem. The styles are enforced via RuboCop.
* There is another gem `rubocop-jekyll` that allows enforcing the same styles, using newer versions of RuboCop, in plugins that have dropped support for Ruby versions older than `2.3.0`.

*Disclaimer: Irrespective of moving this project to Jekyll Org, whether `github-pages` bundles this plugin or not, is outside our control.*

**Comment by parkr** [comment]
@sverrirs Yes, I would ask that you help maintain the project with us! And as @ashmaroli mentioned, the Jekyll Plugins Core Team would be happy to assist in the longer term.

**Comment by MichelZ** [comment]
Please make this happen. Would be a real nice addition to GitHub Pages. Thanks for all the effort!

**Comment by coolbrg** [comment]
I was facing the same issue and then saw this issue and disappointed. 
It is really nice to have.

Right now I have to content myself with `jekyll-paginate`, [Commit ID](https://github.com/coolbrg/coolbrg.github.io/commit/42f51784bd1763e21660349bd94228d42feccadc). 

https://coolbrg.me/blogs/

## When using jekyll-paginate-v2
<img width="1283" alt="Screenshot 2019-05-06 at 6 09 05 AM" src="https://user-images.githubusercontent.com/1132451/57203878-42bd5e00-6fd3-11e9-85c2-69e672bbf68e.png">

**Comment by emma-sax4** [comment]
What's the status of this gem being supported by GitHub Pages? It's so much better than plain old `jekyll-paginate`.

@UnderpantsGnome Do you know if that [hacky suggestion you posted above](https://github.com/sverrirs/jekyll-paginate-v2/issues/9#issuecomment-370189654) compiles on GitHub Pages?

**Comment by UnderpantsGnome** [comment]
@emma-sax4 I don't know if it works on GitHub Pages.

**Comment by gregorriegler** [comment]
no it doesn't :/

**Comment by emma-sax4** [comment]
It looks like it does not work with GitHub Pages (essentially [GitHub pages gem](https://github.com/github/pages-gem) does not include this gem).

**Comment by carlossanlop** [comment]
Hi @sverris , this issue was closed a year ago but the problem still persists. Any chance you could open it again? Maybe shed some light into what the problem is and request help from the community?

**Comment by emmasax4** [comment]
@carlossanlop It's possible you could make a fork of the `github-pages` gem itself, and then add `jekyll-paginate-v2` to it: https://github.com/github/pages-gem/search?q=jekyll-paginate. I'm not sure exactly what your PR would have to look like, but you could start by just adding it as a requirement in every location that this search shows.

**Comment by sverrirs** [comment]
Hi @carlossanlop, the power is solely in the hands of the fine people at Github. Github currently maintains a list of plugins that are allowed to run on github-pages, https://pages.github.com/versions/ This gem simply needs to be added to this list of approved plugins. :)

**Comment by fuzzy76** [comment]
I'm guessing the reach out from @parkr didn't pan out?

**Comment by sverrirs** [comment]
I guess it then unfortunately stranded on me to bring this all the way home.

 This project is now a part of https://www.codeshelter.co/ as I have been hoping to get collaborators to assist with the upkeep as I don't forsee being able to work on this project this full time.

As I understood @parkr then my full commitment was one of the prerequisites to get it approved and I am afraid that due to the popularity I simply don't have that kind of time right now.

**Comment by Akash-Sharma-1** [comment]
Hi,
If you are still facing this issue, you can try writing a Github action for it

I also was facing this issue but then created a Github action that can 'build' my jekyll website and deploy it as github pages. 
Here is the link to the action that I wrote : [jekyll-builder.yml](https://github.com/Akash-Sharma-1/blog/blob/master/.github/workflows/jekyll.yml)

**Basic Idea**
I just build the site and save the build into the `/docs` directory. After this I did setup the github pages to load a static webiste from `/docs` directory itself
Then, in this way, I could build and work with unsupported plugins easily 
(Although, it does take a little more time than the traditional github's jekyll deploying) 

I hope this helps.
Thanks 😊
